internal/appland/client.go

```diff
@@ -84,9 +84,19 @@ func (mapset *MapSet) SetOrganization(org string) *MapSet {
 	return mapset
 }
 
+func (mapset *MapSet) SetBranch(branch string) *MapSet {
+	// only set it if it's not already populated via Git
+	if mapset.Branch == "" {
```

Comment:
Shouldn't the `-b` argument override Git?

Reply:
👍

Reply:
I don't see any reason to facilitate this. If Git determines you're on `master` but you provide an override stating you're on `my-branch`, what's the benefit? The only time this flag should be used is when running `appland upload` within a repository that contains no branch refs. Otherwise, I'd say change Git branches to change this value properly.

Reply:
I would find it surprising that it used `master` when I explicitly asked for `my-branch`. If it's going to work that way (which seems fine), then providing `-b` when there are branch refs should be an error.

Reply:
I would also be fine with this being an error condition. Just don't ignore my flag.